### PR TITLE
Fix find and set default value for unions [20587]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java [hunk 1]
```diff
@@ -35,12 +35,25 @@ else if(type.startsWith("std::vector"))
...
+     *
+     * "If there is a default case specified, the union is initialized to this default case.
+     * In case the union has an implicit default member it is initialized to that case.
+     * In all other cases it is initialized to the first discriminant value specified in IDL"
+     */
+    public static void find_and_set_default_discriminator_value(ArrayList<Definition> defs, UnionTypeCode union_type, String scopeFile, int line)
     {
-        TypeCode dist_type = union_type.getDiscriminator().getTypecode();
         List<Member> members = union_type.getMembers();
+        TypeCode dist_type = union_type.getDiscriminator().getTypecode();
```
Comment (adriancampo) [inline src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:49]: Typo. I think the name should be disc_type.

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java [hunk 2]
```diff
@@ -35,12 +35,25 @@ else if(type.startsWith("std::vector"))
...
-    public static void setUnionDefaultLabel(ArrayList<Definition> defs, UnionTypeCode union_type, String scopeFile, int line)
+    /*!
+     * This function tries to find a default discriminator value for the union following next guidelines extracted from
+     * standard.
+     *
+     * "If there is a default case specified, the union is initialized to this default case.
+     * In case the union has an implicit default member it is initialized to that case.
+     * In all other cases it is initialized to the first discriminant value specified in IDL"
+     */
+    public static void find_and_set_default_discriminator_value(ArrayList<Definition> defs, UnionTypeCode union_type, String scopeFile, int line)
```
Comment (adriancampo) [inline src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:46]: scopeFile and line params are not used, same for checkUnionLabel.
Are they needed?

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java [hunk 3]
```diff
@@ -35,12 +35,25 @@ else if(type.startsWith("std::vector"))
...
     {
-        TypeCode dist_type = union_type.getDiscriminator().getTypecode();
         List<Member> members = union_type.getMembers();
+        TypeCode dist_type = union_type.getDiscriminator().getTypecode();
 
-        if(dist_type != null && union_type.getDefaultMember() != null)
+        if (Kind.KIND_ALIAS == dist_type.getKind())
```
Comment (adriancampo) [inline src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:51]: Will this work if it is the alias of an alias?

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java [hunk 4]
```diff
@@ -35,12 +35,25 @@ else if(type.startsWith("std::vector"))
             return type;
     }
 
-    public static void setUnionDefaultLabel(ArrayList<Definition> defs, UnionTypeCode union_type, String scopeFile, int line)
+    /*!
+     * This function tries to find a default discriminator value for the union following next guidelines extracted from
+     * standard.
+     *
```
Comment (adriancampo) [inline src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:41]: Would be nice to have the section of the standard from where this came from

**Review by adriancampo — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java (hunk 3 above)
Comment (adriancampo) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:51]: NVM, i checked it, it will.

Comment (richiware) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:51]: `getContentTypeCode()` from `AliasTypeCode` is recursive

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java [hunk 5]
```diff
@@ -35,12 +35,25 @@ else if(type.startsWith("std::vector"))
             return type;
     }
 
-    public static void setUnionDefaultLabel(ArrayList<Definition> defs, UnionTypeCode union_type, String scopeFile, int line)
+    /*!
+     * This function tries to find a default discriminator value for the union following next guidelines extracted from
+     * standard.
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java]: Done in 0ffbc6f

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java (hunk 1 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java]: Done in 0ffbc6f

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java (hunk 2 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java]: Old code, removed in 0ffbc6f

File: src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java (hunk 4 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/generator/manager/TemplateUtil.java:41]: Done in 0ffbc6f

**Comment by adriancampo** [comment]
LGTM after CI:
https://github.com/eProsima/Fast-DDS-Gen/actions/runs/8279875339

**Review by adriancampo — APPROVED** [review]

**Pull request merged**